lib/manageiq/release/pull_request_blaster_outer.rb

```diff
@@ -0,0 +1,124 @@
+require 'pathname'
+
+module ManageIQ
+  module Release
+    class PullRequestBlasterOuter
+      attr_reader :repo, :base, :head, :script, :dry_run, :message
+
+      ROOT_DIR = Pathname.new(__dir__).join("..", "..", "..").freeze
+      def initialize(repo, base:, head:, script:, dry_run:, message:)
+        @repo    = repo
+        @base    = base
+        @head    = head
+        @script  = begin
+          s = Pathname.new(script)
+          s = ROOT_DIR.join(script) if s.relative?
+          raise "File not found #{s}" unless File.exist?(s)
+          s.to_s
+        end
+        @dry_run = dry_run
+        @message = message
+      end
+
+      def blast
+        puts "+++ blasting #{repo.github_repo}..."
+
+        repo.git
+        repo.fetch(output: false)
+
+        begin
+          repo.checkout(head, "origin/#{base}")
+        rescue => e
+          raise unless e.message =~ /is not a commit/
+          puts "!!! Skipping #{repo.github_repo}: #{e.message}"
+        end
+
+        run_script
+
+        result = false
+        if !commit_changes
+          puts "!!! Failed to commit changes. Perhaps the script is wrong or #{repo.github_repo} is already updated."
+        elsif dry_run
+          result = "Committed but is dry run"
+        else
+          puts "Do you want to open a pull request on #{repo.github_repo} with the above changes? (Y/N)"
+          answer = $stdin.gets.chomp
+          if answer.upcase.start_with?("Y")
+            fork_repo unless forked?
+            push_branch
+            result = open_pull_request
+          end
+        end
+        puts "--- blasting #{repo.github_repo} complete"
+        result
+      end
+
+      private
+
+      def github
+        ManageIQ::Release.github
+      end
+
+      def forked?
+        github.repos(github.login).any? { |m| m.name == repo.name }
```

Comment:
Perhaps use `git ls-remote` to check if the repo is available since you could have the repo via the API but it might not be pushable yet.